### PR TITLE
doc(helm): remove Tiller reference from the docs

### PR DESCRIPTION
**Pull request by hickeyma**
Partial #5397

### CODE REVIEW

File: docs/chart_template_guide/builtin_objects.md [hunk 1]
```diff
@@ -21,7 +20,6 @@ In the previous section, we use `{{.Release.Name}}` to insert the name of a rele
   - `Capabilities.APIVersions` is a set of versions.
   - `Capabilities.APIVersions.Has $version` indicates whether a version (`batch/v1`) is enabled on the cluster.
   - `Capabilities.KubeVersion` provides a way to look up the Kubernetes version. It has the following values: `Major`, `Minor`, `GitVersion`, `GitCommit`, `GitTreeState`, `BuildDate`, `GoVersion`, `Compiler`, and `Platform`.
-  - `Capabilities.helmVersion` provides a way to look up the Tiller version. It has the following values: `SemVer`, `GitCommit`, and `GitTreeState`.
```
Comment (adamreese) [inline docs/chart_template_guide/builtin_objects.md]: This line is still valid. s/Tiller/Helm/

File: docs/architecture.md [hunk 2]
```diff
@@ -24,41 +24,33 @@ For Helm, there are three important concepts:
...
-
-The Tiller server is also written in Go. It provides a gRPC server to
-connect with the client, and it uses the Kubernetes client library to
-communicate with Kubernetes. Currently, that library uses REST+JSON.
+The Helm client and library is written in the Go programming language.
 
-The Tiller server stores information in ConfigMaps located inside of
-Kubernetes. It does not need its own database.
+The library uses the Kubernetes client library to communicate with Kubernetes. Currently,
+that library uses REST+JSON. Information in ConfigMaps located inside of Kubernetes.
```
Comment (adamreese) [inline docs/architecture.md]:
```suggestion
that library uses REST+JSON. Information in Secrets located inside of Kubernetes.
```

File: docs/chart_best_practices/conventions.md [hunk 3]
```diff
@@ -28,32 +28,12 @@ When SemVer versions are stored in Kubernetes labels, we conventionally alter th
...
 
 - Helm refers to the project, and is often used as an umbrella term
 - `helm` refers to the client-side command
-- Tiller is the proper name of the backend
-- `tiller` is the name of the binary run on the backend
 - The term 'chart' does not need to be capitalized, as it is not a proper noun.
 
 When in doubt, use _Helm_ (with an uppercase 'H').
-
-## Restricting Tiller by Version
```
Comment (adamreese) [inline docs/chart_best_practices/conventions.md]: This section is still valid. s/Tiller/Helm/

File: docs/charts.md [hunk 4]
```diff
@@ -487,8 +486,7 @@ the Kubernetes objects from the charts and all its dependencies are
 
 Hence a single release is created with all the objects for the chart and its dependencies.
 
-The install order of Kubernetes types is given by the enumeration InstallOrder in kind_sorter.go
-(see [the Helm source file](https://github.com/kubernetes/helm/blob/master/pkg/tiller/kind_sorter.go#L26)).
+The install order of Kubernetes types is given by the enumeration InstallOrder in ***TBD***
```
Comment (adamreese) [inline docs/charts.md]: you can use the v3 branch url https://github.com/helm/helm/blob/dev-v3/pkg/tiller/kind_sorter.go#L26

File: docs/charts.md [hunk 5]
```diff
@@ -589,9 +586,8 @@ sensitive_.
   `{{.Files.GetString name}}` functions. You can also access the contents of the file
   as `[]byte` using `{{.Files.GetBytes}}`
 - `Capabilities`: A map-like object that contains information about the versions
-  of Kubernetes (`{{.Capabilities.KubeVersion}}`, Tiller
-  (`{{.Capabilities.HelmVersion}}`, and the supported Kubernetes API versions
```
Comment (adamreese) [inline docs/charts.md]: Keep `{{.Capabilities.HelmVersion}}`

File: docs/install.md [hunk 6]
```diff
@@ -48,7 +45,7 @@ choco install kubernetes-helm
 ## From Script
 
 Helm now has an installer script that will automatically grab the latest version
-of the Helm client and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
+of the Helm and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
```
Comment (adamreese) [inline docs/install.md]:
```suggestion
of Helm and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
```

**Review by adamreese — CHANGES_REQUESTED** [review]
@hickeyma Thanks for this!  Just a few notes.

**Review by adamreese — APPROVED** [review]

**Comment by hickeyma** [comment]
Thanks for merging @adamreese

**Pull request merged**